official/recommendation/neumf_model.py

```diff
@@ -200,28 +200,36 @@ def construct_model(users, items, params):
 
   if params["use_tpu"]:
     with tf.variable_scope("embed_weights", reuse=tf.AUTO_REUSE):
-      cmb_embedding_user = tf.get_variable(
+      cmb_user_latent = tf.keras.layers.Lambda(lambda ids: tf.gather(
+        tf.get_variable(
```

Comment:
If possible, move the call to `tf.get_variable` outside the lambda. It's a bit misleading to call `tf.get_variable` in a Keras layer, since if the Keras model is later called outside the variable scope, `tf.get_variable` will not be affected by the variable scope.